app/models/provider.rb

```diff
@@ -25,14 +25,12 @@ class Provider < ApplicationRecord
 
   supports :refresh_ems
 
-  def self.leaf_subclasses
-    descendants.select { |d| d.subclasses.empty? }
+  def self.concrete_subclasses
+    leaf_subclasses | descendants.select { |d| d.try(:acts_as_sti_leaf_class?) }
```

Comment:
I was expecting the leaf subclasses code to be replaced with an `or` inside the `select`

```suggestion
    descendants.select { |d| d.try(:acts_as_sti_leaf_class?) || d.subclasses.empty? }
```

Reply:
I had the code that way originally, and then I changed it...it felt weird to me to essentially re-implement the leaf_subclasses method (which is what does the `.subclasses.empty?` and comes from more_core_extensions).  I thought keeping leaf_subclasses as is was clearer, and then concrete_subclasses is like a refined version of that.

While I named it `acts_as_sti_leaf_class`, I have no problem changing the name, but wanted something that was more generic.